assets/js/blocks/products/all-products/block.js

```diff
@@ -31,6 +31,7 @@ class Block extends Component {
 		return (
 			<ProductListContainer
 				attributes={ attributes }
+				blockName="woocommerce/all-products"
```

Comment:
Good idea! Done in fb8c469aade450d9b3e8ac29f4a2f8f765d9e1b1.

> As I write this, I realize there may be other things we're adding here that would work better with context (possibly layout_template?) and ideally the extensibility is something added either via hoc or context so we can just wrap components we want to make extensible.

For now, I only moved `blockName` to the context in order to get feedback, but I agree some other things could be moved to the context as well. Would make it sense doing that in follow-ups instead of directly here?

Reply:
👍 looks great but I do have some suggestions regarding tweaking what is exposed for the context.

> Would make it sense doing that in follow-ups instead of directly here?

Yup I agree, we can work on that in subsequent pulls (if necessary).